src/d3a/d3a_core/sim_results/kpi.py

```diff
@@ -76,10 +76,12 @@ def _accumulate_self_consumption_buffer(self, trade):
             self.self_consumption_buffer_wh += trade.offer.energy * 1000
 
     def _dissipate_self_consumption_buffer(self, trade):
-        if trade.seller_origin in self.ess_list and trade.seller_origin == trade.seller:
+        if trade.seller_origin in self.ess_list:
             # self_consumption_buffer needs to be exhausted to total_self_consumption
             # if sold to internal consumer
-            if trade.buyer_origin in self.consumer_list and self.self_consumption_buffer_wh > 0:
+            if trade.buyer_origin in self.consumer_list and \
+                    trade.buyer_origin == trade.buyer and \
```

Comment:
So there was a double counting and you added a check to only track the trade that happens in the market where the device is connected to?

Reply:
As I mentioned in the reply to Faizan, the tracking is now done in the buyer side, there was a check to avoid double counting the same offer in the seller side before, but the tracking needed to be done on the buyer side for it to be successful.